server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/MySQLAffectedRowsExceedLimit.java

```diff
@@ -120,14 +124,51 @@ public List<DialectType> getSupportsDialectTypes() {
         return Arrays.asList(DialectType.MYSQL, DialectType.OB_MYSQL);
     }
 
+    /**
+     * MySQL checks the count of value list. Process mode: case1: For INSERT INTO ... VALUES(...)
+     * statements, ODC checks the count of value list. case2: For INSERT INTO ... SELECT ... statements,
+     * ODC runs EXPLAIN statements to get affected rows.
+     *
+     * @param sql target sql
+     * @return affected rows
+     */
+    private long getMySqlAffectedRowsByCount(String sql) {
+
+        sql = sql.trim().replace("\n", "").toUpperCase();
+        String valuesRegex = "\\)\\s*VALUES?\\s*\\(";
+        String selectRegex = "\\)\\s*SELECT";
+
+        Pattern valuesPattern = Pattern.compile(valuesRegex);
```

Comment:
we may compile them at static code block:
```java
private static Pattern VALUES_PATTERN = Pattern.compile(valuesRegex);
private static Pattern SELECT_PATTERN = Pattern.compile(selectRegex);
```

Reply:
got it. It's best to use sql-parser, I didn't know it could be used like this before.